AutoRaise.mm

```diff
@@ -334,6 +334,129 @@ - (void)onTick:(NSNumber *)timerInterval {
     oldPoint.x = oldPoint.y = 0;
 }
 
+//----------------------------------------------configuration-----------------------------------------------
+
+NSMutableDictionary *parameters = [[NSMutableDictionary alloc] init];
+NSArray   *parametersDictionary = @[@"delay", @"warpX", @"warpY", @"scale"];
+
+@interface ConfigClass:NSObject
+- (NSString *) getFilePath: (NSString *) filename;
+- (BOOL) fileExist: (NSString *) filename;
+- (void) readCommandLineConfig: (int) argc;
+- (void) readOriginalConfig;
+- (void) readHiddenConfig;
+- (void) validateParameters;
+@end
+
+@implementation ConfigClass
+
+- (NSString *) getFilePath: (NSString *) filename {
+    filename = [NSString stringWithFormat: @"%@/%@", NSHomeDirectory(), filename];
+    if (not [[NSFileManager defaultManager] fileExistsAtPath: filename]) { filename = @""; }
```

Comment:
I understand better now your use cases for getFilePath and fileExist. What you really want, is to get the file path, and if it not exists, return something that indicates that. This can be done with NULL, so you would say:

    filename = NULL;

Then you can drop fileExists, and in the rest of the code do things like:

    NSString * filename = getFilePath(@"some file");
    if (filename) { ... } else { ... }

Reply:
> Then you can drop fileExists, and in the rest of the code do things like:
> 
> ```
> NSString * filename = getFilePath(@"some file");
> if (filename) { ... } else { ... }
> ```

Are we talking about [NSString's -boolValue]?

> Declaration
> @property(readonly) BOOL boolValue;
> Discussion
> This property is YES on encountering one of "Y", "y", "T", "t", or a digit 1-9—the method ignores any trailing characters. This property is NO if the receiver doesn’t begin with a valid decimal text representation of a number.
> 

so for
```
NSString * filename = getFilePath(@"some real existing file");
[@"existing file path" -boolValue] => false 
```
:(

It is possible to replace fileExist with condition like this:
`(not [[self getFilePath: delayFile] isEqual: NULL] )`
but it seems less readable. hm?